File: Majority Element/Majority_Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        element, cnt = 0, 0
        for i in nums:
            if element == i:
                cnt += 1
            else:
                cnt -= 1
            if cnt <= 0:
                element = i
                cnt = 1
        return element
        

File: Majority Element/Majority_Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int element = 0, cnt = 0;
        for (int i : nums) {
            if (element == i) cnt++;
            else cnt--;
            if (cnt <= 0) {
                element = i;
                cnt = 1;
            }
        }
        return element;
    }
}